pkg/taskutil/spec.go

```diff
@@ -166,15 +179,34 @@ func TreeFromDefaultPlan(opts DefaultPlanOptions) (interface{}, error) {
 		return nil, fmt.Errorf("unknown scheme %s in spec URL", u.Scheme)
 	}
 
-	r, err := loader.LoadSpec()
-	if err != nil || r == nil {
-		return nil, err
-	}
-
 	var env struct {
 		Value transfer.JSONInterface `json:"value"`
 	}
-	if err := json.NewDecoder(r).Decode(&env); err != nil {
+
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	waitOptions := []retry.WaitOption{}
+
+	err = retry.Wait(ctx, func(ctx context.Context) (bool, error) {
+		r, rerr := loader.LoadSpec()
+		if rerr != nil {
+			return false, rerr
+		}
+
+		if r == nil {
+			return true, nil
+		}
+
+		if derr := json.NewDecoder(r).Decode(&env); derr != nil {
+			return false, derr
+		}
+
+		return true, nil
+	}, waitOptions...)
```

Comment:
FWIW, if you have no values to pass for a varargs argument, you can omit it entirely. I.e., this code is equivalent to `retry.Wait(ctx, fn)`.

Reply:
Naturally. As a next step, the wait options will likely be definable as options. In the meantime, I find this more clear and explicit to anyone reading or updating this code.